### PR TITLE
dev-vcs/git-cola: add 4.8.2 

### PR DESCRIPTION
**Pull request by stkw0**
This PR will leave the next PR obsolete: https://github.com/gentoo/gentoo/pull/35653 

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Comment by bearsh** [comment]
Thanks @stkw0 for creating this PR.

Shouldn't we drop pyqt5/pyside2 as per gentoo qt policies (https://wiki.gentoo.org/wiki/Project:Qt/Policies)? or at least default to a qt6 backend?

**Comment by stkw0** [comment]
I thought about it. If it was a pure qt I would prefer to have an IUSE="qt6" and then  `qt6? ( ${QT6_DEPS} )` and `!qt6? ( ${QT5_DEPS} )` but having pyqt and pyside it would be confusing.  For the defaults I followed QtPy which enables pyqt5 by default, but if you think pyqt6 should be the default I will change it.

**Comment by bearsh** [comment]
I see. And it makes sense to follow qtpy's defaults...

File: dev-vcs/git-cola/git-cola-4.8.2.ebuild [hunk 1]
```diff
@@ -14,14 +14,22 @@ SRC_URI="https://github.com/${PN}/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
...
 	$(python_gen_cond_dep '
 		dev-python/numpy[${PYTHON_USEDEP}]
+		dev-python/polib[${PYTHON_USEDEP}]
 		dev-python/pygments[${PYTHON_USEDEP}]
-		dev-python/PyQt5[${PYTHON_USEDEP},gui,widgets]
-		dev-python/QtPy[gui,network,${PYTHON_USEDEP}]
+		pyqt5? ( dev-python/QtPy[pyqt5,gui,network,${PYTHON_USEDEP}] )
+		pyqt6? ( dev-python/QtPy[pyqt6,gui,network,${PYTHON_USEDEP}] )
+		pyside2? ( dev-python/QtPy[pyside2,gui,network,${PYTHON_USEDEP}] )
+		pyside6? ( dev-python/QtPy[pyside6,gui,network,${PYTHON_USEDEP}] )
```
[the remark this answers is not in the document]
Comment (ionenwks) [reply dev-vcs/git-cola/git-cola-4.8.2.ebuild:32]: Also, can't blame you for doing so, but plesae don't use QtPy as a reference, its handling of the situation is questionable and been causing problems.

[the remark this answers is not in the document]
Comment (ionenwks) [reply dev-vcs/git-cola/git-cola-4.8.2.ebuild:32]: On another note, we'll probably make USE=pyqt6 a default on QtPy at least so it won't be annoying for users here.

[the remark this answers is not in the document]
Comment (stkw0) [reply dev-vcs/git-cola/git-cola-4.8.2.ebuild]: Seems to work with pyqt6, so I forced pyqt6 unconditionally as requested

File: dev-vcs/git-cola/git-cola-4.8.2.ebuild [hunk 2]
```diff
@@ -63,5 +65,10 @@ src_compile() {
 
 src_install() {
 	distutils-r1_src_install
+
+	# patch the binaries to use desired qtpy backend
+	local qt_api="pyqt6"
+	sed -i "s|import sys|import sys\nimport os\nos.environ['QT_API'] = '${qt_api}'\n|" "${D}"/usr/bin/* || die
```
Comment (negril) [inline dev-vcs/git-cola/git-cola-4.8.2.ebuild:71]:
```suggestion
	sed -i "s|import sys|import sys\nimport os\nos.environ['QT_API'] = 'pyqt6'\n|" "${D}"/usr/bin/* || die
```

File: dev-vcs/git-cola/git-cola-4.8.2.ebuild [hunk 3]
```diff
@@ -44,6 +44,8 @@ distutils_enable_tests pytest
 
 src_prepare() {
 	sed -i "s|doc/git-cola =|doc/${PF} =|" setup.cfg || die
+	# remove bundled qtpy and polib
+	rm -Rf "${S}"/qtpy "${S}"/cola/polib.py || die
```
Comment (negril) [inline dev-vcs/git-cola/git-cola-4.8.2.ebuild:48]:
```suggestion
	rm -Rf qtpy cola/polib.py || die
```

`src_prepare` is run in `S`.

**Review by negril — CHANGES_REQUESTED** [review]
Some cosmetical clean-ups.

File: dev-vcs/git-cola/git-cola-4.8.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+BDEPEND="
+	sys-devel/gettext
+	$(python_gen_cond_dep "
+		dev-python/setuptools-scm[\${PYTHON_USEDEP}]
+		test? (
+			${VIRTUALX_DEPEND}
+			dev-python/pytest[\${PYTHON_USEDEP}]
+			dev-python/QtPy[\${PYTHON_USEDEP},pyqt6,gui,network]
+		)
+	")
```
[the remark this answers is not in the document]
Comment (stkw0) [reply dev-vcs/git-cola/git-cola-4.8.2.ebuild]: Thanks! All suggestions applied

File: dev-vcs/git-cola/git-cola-4.8.2.ebuild [hunk 5]
```diff
@@ -14,14 +14,14 @@ SRC_URI="https://github.com/${PN}/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
 
 LICENSE="GPL-2"
 SLOT="0"
-KEYWORDS="~amd64 ~x86"
+KEYWORDS="~amd64"
```
Comment (a17r) [inline dev-vcs/git-cola/git-cola-4.8.2.ebuild:17]: dev-python/PyQt6 has just been ~x86 keyworded, so you should be able to keep this.

**Review by a17r — CHANGES_REQUESTED** [review]

File: dev-vcs/git-cola/git-cola-4.8.2.ebuild (hunk 5 above)
Comment (stkw0) [reply dev-vcs/git-cola/git-cola-4.8.2.ebuild:17]: Seems there still some issues with ~x86

Comment (a17r) [reply dev-vcs/git-cola/git-cola-4.8.2.ebuild:17]: what issues?

Comment (stkw0) [reply dev-vcs/git-cola/git-cola-4.8.2.ebuild:17]: pkgcheck complanied about QtPy, but seems it was even stable on x86, so I don't really understand. In any case, I think x86 was being slowly deprecated, so didn't wasted too much investigating it.

Comment (a17r) [reply dev-vcs/git-cola/git-cola-4.8.2.ebuild:17]: If you depended on PyQt6 via QtPy[pyqt6], but did not rebase this branch onto a recent pull of origin/master, then failure on x86 would have been clear. It was unmasked only a couple of weeks ago on that branch (and keyworded only shortly before).

File: dev-vcs/git-cola/git-cola-4.8.2.ebuild (hunk 1 above)
[the remark this answers is not in the document]
Comment (stkw0) [reply dev-vcs/git-cola/git-cola-4.8.2.ebuild:32]: Shit, sorry. I was working in two different computers. Seems I switched computer and made a small change thinking I was in the correct state and force pushed the change. Effectively removing the changes I made on the other computer...

[the remark this answers is not in the document]
Comment (a17r) [reply dev-vcs/git-cola/git-cola-4.8.2.ebuild:32]: You could still try and pull your changes out of git reflog, at least on a system that contained that state of your work.

[the remark this answers is not in the document]
Comment (stkw0) [reply dev-vcs/git-cola/git-cola-4.8.2.ebuild:32]: Yeah, that's what I usually do. It was mostly a problem of being in a rush and not properly checking I pulled the last changed, not a technical one.

**Comment by negril** [comment]
You also seem to have lost all the other changes you made in between with https://github.com/gentoo/gentoo/compare/c102479815f6fcbd78130620b2f5897409780135..34e06773b9fb87fd934d03989ab15724efef7f2f

**Comment by stkw0** [comment]
Maybe I should do another PR and fix my mess?

**Comment by a17r** [comment]
yes, please.

**Comment by stkw0** [comment]
Done: https://github.com/gentoo/gentoo/pull/38853

**Pull request closed without merging**